src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -638,12 +640,9 @@ export interface ITerminalInstance {
 	/**
 	 * Dispose the terminal instance, removing it from the panel/service and freeing up resources.
 	 *
-	 * @param immediate Whether the kill should be immediate or not. Immediate should only be used
-	 * when VS Code is shutting down or in cases where the terminal dispose was user initiated.
-	 * The immediate===false exists to cover an edge case where the final output of the terminal can
-	 * get cut off. If immediate kill any terminal processes immediately.
+	 * @param isShutdown Whether the kill was triggered by lifecycle shutdown
 	 */
-	dispose(immediate?: boolean): void;
```

Comment:
Is `immediate` deprecated? There was only one placed that used it (in `terminalQuickAccess.ts`) but I changed it to `terminalService.safeDisposeTerminal` so it's the same as other calls.
If not I'll revet this change.

Reply:
I'm not totally sure what the current state of this is, but it was always a bit of a hack to add a param to dispose like this so I'd want to move away from it if we were to change things in this area.